stdlib/src/memory/memory.mojo

```diff
@@ -49,30 +49,49 @@ fn _memcmp_impl(s1: DTypePointer, s2: __type_of(s1), count: Int) -> Int:
         for i in range(count):
             var s1i = s1[i]
             var s2i = s2[i]
-            var diff = s1i - s2i
-            if diff:
-                return 1 if diff > 0 else -1
+            var smaller = s1i < s2i
+            var bigger = s1i > s2i
+            if smaller or bigger:
+                return -int(smaller) + int(bigger)
```

Comment:
Yeah, that is even simpler, thanks!

Reply:
Can we do similar things to `SIMD`?

Reply:
I actually realised why I did it the way I did before I wanted to make a branchless implementation, but stoped half way :)
```
  var result = 0
  var i = 0
  while i < count:
    var s1i = s1[i]
    var s2i = s2[i]
    var smaller = s1i < s2i
    var bigger = s1i > s2i
    i += 1 + count * int(smaller or bigger)
    result = -1 * int(smaller) + 1 * int(bigger)
  return result
```
This would be branchless implementation, is a bit more complex but my guess it should be faster. Anyways let's go with the simple one for now. I will create a gist for myself. When we have benchmarks infrastructure setup we can do such micro optimisations.

Reply:
Somewhat relevant [reference](https://community.arm.com/arm-community-blogs/b/infrastructure-solutions-blog/posts/porting-x86-vector-bitmask-optimizations-to-arm-neon). I still think we could these to a `cmp` function in `SIMD` so you can use as much trick as possible.

Reply:
@soraros you were right regarding simplifying SIMD part, I just pushed an update.